include/treelite/tree.h

```diff
@@ -405,10 +419,43 @@ static_assert(std::is_standard_layout<ModelParam>::value,
 inline void InitParamAndCheck(ModelParam* param,
                               const std::vector<std::pair<std::string, std::string>>& cfg);
 
+class Model {
```

Comment:
Consider moving the refactoring of `struct Model` into `class Model` and `class ModelImpl` (without template parameters) into a separate pull request.

Reply:
Will create a separate PR.

Reply:
#201 has been merged.